# Analyzing CIA Factbook Data Using SQL

In this project we'll be working with data from the CIA World Factbook which is a detailed collection of data with statistics about all the countries on Earth. The factbook contains demographic information like population, population growth, and total land and water area. We'll be working with SQL to analyze data from this database. 

In [6]:
%%capture
%load_ext sql
%sql sqlite:///factbook.db

'Connected: None@factbook.db'

The code above simply connects our jupyter notebook to the factbook database file, the flavor of SQL we'll be using is sqlite.

In [7]:
%%sql
SELECT *
    FROM sqlite_master
    WHERE type='table';

Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

In [8]:
%%sql
SELECT *
    FROM facts
    LIMIT 5;

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

# Column descriptions

- name: the name of the country
- area: the country's total area (both land and water)
- area_land: the country's land area (in square kilometers)
- area_water: the country's waterarea (in square kiometers)
- population: the country's population
- population_growth: the country's population growth as a percentage
- birth_rate: the country's birth rate, or number of births per year per 1000 people
- death_rate: the country's death rate, or number of death per year per 1000 people

# Summary Statistics for Population & Population_Growth

In [9]:
%%sql 

SELECT MIN(population), MAX(population), MIN(population_growth), MAX(population_growth)
    FROM facts;

Done.


[(0, 7256490011, 0.0, 4.02)]

We can clearly see we have some missing values in the database since the minimum population and minimum population_growth are 0. Along with those two 0 values, the maximum population in the dataset is 7,256,490,011 - that seems just around where the entire worlds population is at these days.

In [12]:
%%sql 
SELECT *
    FROM facts
    WHERE population == (SELECT MIN(population)
                            FROM facts
                        );

Done.


[(250, 'ay', 'Antarctica', None, 280000, None, 0, None, None, None, None)]

In [13]:
%%sql
SELECT *
    FROM facts
    WHERE population == (SELECT MAX(population)
                            FROM facts
                        );

Done.


[(261, 'xx', 'World', None, None, None, 7256490011, 1.08, 18.6, 7.8, None)]

Above we used subqueries to take a look at first; the country with the minimum population which we stated was 0 and the country is Antarctica. Next, we used subqueries again to take a look at the country with the maximum population value which was over 7 billion and we can see that it is indeed the number for the entire worlds population.